runtime/lua/vim/fs.lua

```diff
@@ -148,8 +148,17 @@ function M.dir(path, opts)
   vim.validate('path', path, 'string')
   vim.validate('depth', opts.depth, 'number', true)
   vim.validate('skip', opts.skip, 'function', true)
+  vim.validate('filter', opts.filter, 'function', true)
   vim.validate('follow', opts.follow, 'boolean', true)
 
+  if opts.skip ~= nil then
+    vim.deprecate('opts.skip', 'opts.filter', '0.13')
```

Comment:
This is one of those things that might just stick around. Don't need to rush its removal.

```suggestion
    vim.deprecate('opts.skip', 'opts.filter', '1.0')
```